pkg/services/ngalert/remote/alertmanager_test.go

```diff
@@ -315,23 +329,38 @@ func TestCompareAndSendConfiguration(t *testing.T) {
 			},
 			"",
 		},
+		{
+			"no error, with auto-generated routes",
+			strings.Replace(testGrafanaConfigWithSecret, `"password":"test"`, fmt.Sprintf("%q:%q", "password", base64.StdEncoding.EncodeToString(testValue)), 1),
+			testAutogenFn,
+			&client.UserGrafanaConfig{
+				GrafanaAlertmanagerConfig: cfgWithAutogenRoutes,
+			},
+			"",
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(tt *testing.T) {
-			am, err := NewAlertmanager(cfg,
+			ctx := context.Background()
+			am, err := NewAlertmanager(ctx,
+				cfg,
 				fstore,
 				decryptFn,
-				test.autogenFn,
+				NoopAutogenFn,
 				m,
 				tracing.InitializeTracerForTest(),
 			)
 			require.NoError(t, err)
 
+			// Adding the autogenFn after creating the Alertmanager
+			// to simulate errors when comparing the configuration.
+			am.autogenFn = test.autogenFn
```

Comment:
If we add this when creating the AM struct, the error (if any) will cause that creation to fail.
We want to test what happens if the autogen returns an error during `CompareAndSendConfiguration()` fails, not `NewAlertmanager()`.